### PR TITLE
[CDF-25745] 👮 Add request tracker

### PR DESCRIPTION
**Pull request by doctrino**
# Description

If you do a request with 1000 items in the `HTTPCLient`, the current implementation risks doing 1999 requests to fail each individual item. This PR introduces a maximum threshold for the number of failed request before the HTTP Client gives up. 

## Changelog

- [ ] Patch
- [x] Skip

### CODE REVIEW

**Review by Aashutosh-cognite — APPROVED** [review]
LGTM

File: tests/test_unit/test_cdf_tk/test_utils/test_http_client.py [hunk 1]
```diff
@@ -403,6 +404,69 @@ def test_timeout_error(self, http_client_one_retry: HTTPClient, rsps: responses.
             FailedRequestItem(id=1, error="RequestException after 1 attempts (read error): Simulated timeout error")
         ]
 
+    @pytest.mark.usefixtures("disable_gzip")
```
Comment (rsjr) [inline tests/test_unit/test_cdf_tk/test_utils/test_http_client.py:407]: Can you add a test case to cover (edge) case when max_failures_before_abort is 0?

**Review by Aashutosh-cognite — APPROVED** [review]

**Review by rsjr — APPROVED** [review]
risk review ok
